samples/net/common/net_sample_common.c

```diff
@@ -19,7 +19,11 @@ static void l4_event_handler(struct net_mgmt_event_callback *cb, uint32_t event,
 			     struct net_if *iface)
 {
 	switch (event) {
+#if defined(CONFIG_MODEM_HL7800) && !defined(CONFIG_DNS_SERVER_IP_ADDRESSES)
```

Comment:
I am not a big fan of adding hw specific config options to generic sample. What is this change trying to achieve, it is not very clear?

Reply:
The last thing the HL7800 does is get the DNS address from the network and reconfigure the DNS resolver. Waiting for the DNS address ensures the network interface is ready for operation.

Reply:
Ok, but we could make this more generic by introducing a Kconfig option for it.
```
config NET_SAMPLE_WAIT_DNS_SERVER_ADDITION
	bool "Wait DNS server addition before considering connection to be up"
	depends on MODEM_HL7800 && !DNS_SERVER_IP_ADDRESSES
	help
	  Make sure we get DNS server addresses from the network
	  before considering the connection to be up.
```
and then use that option in sources. This way we avoid adding board specific options to generic sample, and also allow other boards to add "depends on" to the kconfig if needed.

Reply:
I'm a bit puzzled how does this even work given that the event mask (`L4_EVENT_MASK`) was not reconfigured to cover `NET_EVENT_DNS_SERVER_ADD`? Plus there's a similar change in the LwM2M sample.

Reply:
> I'm a bit puzzled how does this even work given that the event mask (`L4_EVENT_MASK`) was not reconfigured to cover `NET_EVENT_DNS_SERVER_ADD`? Plus there's a similar change in the LwM2M sample.

Very good point. I'll look into this. I tested many time and the sample worked great... Maybe I got lucky with timing.

Reply:
https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/net/ip/net_mgmt.c#L203 - indeed, it seems net mgmt implementation is built on assumption net events "command" part is a bitfield, which isn't really the case... And changing  the command representation doesn't seem like a viable option given there're only 16 bits for command in the event mask (and for example there's more than 20 IPv6 events)...

Reply:
> Ok, but we could make this more generic by introducing a Kconfig option for it.
> 
> ```
> config NET_SAMPLE_WAIT_DNS_SERVER_ADDITION
> 	bool "Wait DNS server addition before considering connection to be up"
> 	depends on MODEM_HL7800 && !DNS_SERVER_IP_ADDRESSES
> 	help
> 	  Make sure we get DNS server addresses from the network
> 	  before considering the connection to be up.
> ```
> 
> and then use that option in sources. This way we avoid adding board specific options to generic sample, and also allow other boards to add "depends on" to the kconfig if needed.

I implemented Kconfig and made this more generic across the 3 samples.
I tested all 3 samples again on a live cell network instead of my callbox. They work great.

Reply:
> I'm a bit puzzled how does this even work given that the event mask (`L4_EVENT_MASK`) was not reconfigured to cover `NET_EVENT_DNS_SERVER_ADD`? Plus there's a similar change in the LwM2M sample.

I fixed this with the new kconfig option.